controllers/nstemplatetierrevisioncleanup/nstemplatetier_revision_cleanup_controller_test.go

```diff
@@ -0,0 +1,276 @@
+			//then
+			require.EqualError(t, err, "unable to delete the current Tier Template Revision base1ns-clusterresources-123456new-ttrcr: some error cannot delete")
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+		})
+		t.Run("tier label not found", func(t *testing.T) {
+			ttr.Labels = map[string]string{}
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr)
+			//when
+			res, err := r.Reconcile(context.TODO(), req)
+			//then
+			require.NoError(t, err)
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).DoNotExist()
+		})
+
+		t.Run("tier label not found but error deleting ttr", func(t *testing.T) {
+			ttr.Labels = map[string]string{}
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr)
+			cl.MockDelete = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.DeleteOption) error {
+				return fmt.Errorf("some error cannot delete")
+			}
+			//when
+			res, err := r.Reconcile(context.TODO(), req)
+			//then
+			require.EqualError(t, err, "unable to delete the current Tier Template Revision base1ns-clusterresources-123456new-ttrcr: some error cannot delete")
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+		})
+
+	})
+
+}
+
+func prepareReconcile(t *testing.T, name string, initObjs ...runtimeclient.Object) (*nstemplatetierrevisioncleanup.Reconciler, reconcile.Request, *test.FakeClient) {
+	os.Setenv("WATCH_NAMESPACE", test.HostOperatorNs)
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	cl := test.NewFakeClient(t, initObjs...)
+	r := &nstemplatetierrevisioncleanup.Reconciler{
+		Client: cl,
+	}
+	return r, reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: operatorNamespace,
+		},
+	}, cl
+}
+func createttr(nsTTier toolchainv1alpha1.NSTemplateTier, name string, crtime metav1.Time) *toolchainv1alpha1.TierTemplateRevision {
+	labels := map[string]string{
+		toolchainv1alpha1.TierLabelKey: nsTTier.Name,
+	}
+	ttr := &toolchainv1alpha1.TierTemplateRevision{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:         operatorNamespace,
+			Name:              name,
+			Labels:            labels,
+			CreationTimestamp: crtime,
+		},
+		Spec: toolchainv1alpha1.TierTemplateRevisionSpec{
+			Parameters: nsTTier.Spec.Parameters,
+		},
+	}
+	return ttr
+}
+
+func createSpace(nsTTier *toolchainv1alpha1.NSTemplateTier) *toolchainv1alpha1.Space {
+	testSpace := spacetest.NewSpace(test.HostOperatorNs, "oddity1",
+		spacetest.WithTierNameAndHashLabelFor(nsTTier),
+		spacetest.WithSpecTargetCluster("member-1"),
+		spacetest.WithStatusTargetCluster("member-1"),
+		spacetest.WithFinalizer(),
+		spacetest.WithCondition(spacetest.Ready()))
```

Comment:
it doesn't seem that we actually need those for the sake of the test
```suggestion
		spacetest.WithTierNameAndHashLabelFor(nsTTier))
```